core/src/test/java/org/infinispan/notifications/cachelistener/CacheNotifierImplWithConverterTest.java

```diff
@@ -0,0 +1,322 @@
+      CacheEntryPassivatedEvent event = (CacheEntryPassivatedEvent) cl.getEvents().get(0);
+      assert event.getKey().equals("k");
+      assert event.getValue().equals("custom event for test (CACHE_ENTRY_PASSIVATED,k,v)");
+      assert cl.getEvents().get(1).getCache() == mockCache;
+      assert cl.getEvents().get(1).getType() == Event.Type.CACHE_ENTRY_PASSIVATED;
+      event = (CacheEntryPassivatedEvent) cl.getEvents().get(1);
+      assert event.getKey().equals("k");
+      assert event.getValue().equals("custom event for test (CACHE_ENTRY_PASSIVATED,k,v)");
+   }
+
+   public void testNotifyTransactionCompleted() {
+      GlobalTransaction tx = mock(GlobalTransaction.class);
+      n.notifyTransactionCompleted(tx, true, ctx);
+      n.notifyTransactionCompleted(tx, false, ctx);
+
+      assert cl.getInvocationCount() == 2;
+      assert cl.getEvents().get(0).getCache() == mockCache;
+      assert cl.getEvents().get(0).getType() == Event.Type.TRANSACTION_COMPLETED;
+      assert ((TransactionCompletedEvent) cl.getEvents().get(0)).isTransactionSuccessful();
+      assert ((TransactionCompletedEvent) cl.getEvents().get(0)).getGlobalTransaction() == tx;
+      assert cl.getEvents().get(1).getCache() == mockCache;
+      assert cl.getEvents().get(1).getType() == Event.Type.TRANSACTION_COMPLETED;
+      assert !((TransactionCompletedEvent) cl.getEvents().get(1)).isTransactionSuccessful();
+      assert ((TransactionCompletedEvent) cl.getEvents().get(1)).getGlobalTransaction() == tx;
+   }
+
+   public void testNotifyTransactionRegistered() {
+      GlobalTransaction tx = mock(GlobalTransaction.class);
+      n.notifyTransactionRegistered(tx, false);
+      n.notifyTransactionRegistered(tx, false);
+
+      assert cl.getInvocationCount() == 2;
+      assert cl.getEvents().get(0).getCache() == mockCache;
+      assert cl.getEvents().get(0).getType() == Event.Type.TRANSACTION_REGISTERED;
+      assert ((TransactionRegisteredEvent) cl.getEvents().get(0)).getGlobalTransaction() == tx;
+      assert cl.getEvents().get(1).getCache() == mockCache;
+      assert cl.getEvents().get(1).getType() == Event.Type.TRANSACTION_REGISTERED;
+      assert ((TransactionRegisteredEvent) cl.getEvents().get(1)).getGlobalTransaction() == tx;
+   }
+   
+   @NamedFactory(name = "to-string-converter-factory")
+   public class ToStringEventConverterFactory implements CacheEventConverterFactory {
+
+   	@Override
+   	public CacheEventConverter<String, String, String> getConverter(Object[] params) {
+   		// TODO Auto-generated method stub
+   		return new ToStringConverter();
+   	}
+   	class ToStringConverter implements CacheEventConverter<String, String, String>  {
+   		      public String convert(String key, String oldValue, Metadata oldMetadata, 
+   		               String newValue, Metadata newMetadata, EventType eventType) {
+   		         return "custom event for test ("+(eventType==null ? "null" : eventType.getType().name())+","+ key+","+newValue+")";
```

Comment:
Shouldn't the eventType always be non null now? I say throwing an exception would be better so the test fails.